pkg/azurefile/utils.go

```diff
@@ -234,7 +234,7 @@ func SetVolumeOwnership(path, gid, policy string) error {
 		return fmt.Errorf("convert %s to int failed with %v", gid, err)
 	}
 	gidInt64 := int64(id)
-	fsGroupChangePolicy := v1.FSGroupChangeAlways
+	fsGroupChangePolicy := v1.FSGroupChangeOnRootMismatch
```

Comment:
thanks for raising the concern. currently `fsGroupChangePolicy` is an azure file volume property, I think we could do some backwards compatibility with future version of CSI and k8s these changes are finalized, we already hit some mount issues recently and need to workaround ASAP. thanks.

Name | Meaning | Example | Mandatory | Default value 
--- | --- | --- | --- | ---
fsGroupChangePolicy | indicates how volume's ownership will be changed by the driver, pod `securityContext.fsGroupChangePolicy` is ignored  | `OnRootMismatch`(by default), `Always`, `None` | No | `OnRootMismatch`

Reply:
How are these properties of CSI driver? This code directly uses `v1.FSGroupChangeOnRootMismatch` from core k8s type, where intended purpose of that type is entirely different. 

Can you elaborate on what backward compatibility issue this PR fixes? I thought NFS support within azurefile driver is relatively new feature.

Reply:
these values are for `volume.SetVolumeOwnership`, NFS feature is already released for around 1 year, we cannot remove `volume.SetVolumeOwnership` inside the driver now since it may break some scenarios, and we hit some edge cases that mount would fail when there are large num of files inside one volume, in such case, we ask user to set `fsGroupChangePolicy: None` to mitigate the slow mount or mount failure issue.

let me remove the dependency on `v1.FSGroupChangeOnRootMismatch`, we should use `OnRootMismatch`, `Always`, `None` strings directly.